src/charm.py

```diff
@@ -65,18 +146,37 @@ def _on_alertmanager_changed(self, event):
         self._stored.alertmanager_port = port
         self._stored.alertmanagers = addrs
 
-        self._configure_pod()
+        self._on_config_changed(event)
```

Comment:
not sure if it is better to just emit a ConfigChangedEvent instead

Reply:
I do not see the benefit of one event handler to be raising another event, and the charm to then waiting for the next iteration of Juju invoking the same charm's code, for the so raised event to be then handled. Isn't it simpler if all that needs to done to handle the present event is done, even if this means invoking another function of the same charm, even if the other function is event handler ? Besides there is no reason to defer the event here either.

Reply:
I did not realize an emitted event is deferred

Reply:
As far as I understand it will go into the queue of events maintained by Juju, and will be scheduled for forwarding to the charm.